### PR TITLE
introduce `DiMuonMassBiasClient` and use in `ALCARECOTkAlDiMuonAndVertexDQM`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

This PR is a follow-up to PR https://github.com/cms-sw/cmssw/pull/39128 and adds a client `DQMEDHarvester` to create profiles of the mass bias vs track kinematics, based on the 2D `MonitorElement`s introduced there.
I profit of this PR to introduce a bug-fix in commit 2456648988a74bb8907702dd13e299f0b3ad5a59 in order to protect vertex tool compatibility failures with a try/catch mechanism. 
Running privately on data the following exception was observed if not protected:

https://github.com/cms-sw/cmssw/blob/1ce6a20d05a96a0a336eb63e2b3c56b7c028c3ef/RecoVertex/VertexTools/src/VertexDistance3D.cc#L57-L60

#### PR validation:

Run the (augmented) unit tests of this package and also run manually on 2022C data obtaining the fits at this [link](http://musich.web.cern.ch/musich/display/mergedFitsData.pdf).
The resulting DQM file has been uploaded to a private GUI that is available by:

```
 ssh -NL 8060:localhost:8060 <USER>@lxplus724.cern.ch
```

and visiting:

https://tinyurl.com/2pwdox97

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, but will need to be backported.

### CODE REVIEW

**Comment by mmusich** [comment]
type bug-fix, new-feature, trk

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by emanueleusai** [comment]
+1

**Comment by qliphy** [comment]
+1

**Comment by makortel** [comment]
This PR caused many workflows to fail in IB multithreaded tests because `DiMuonMassBiasClient` does not support concurrent lumis, see #39180.

File: DQMOffline/Alignment/src/DiMuonMassBiasMonitor.cc [hunk 1]
```diff
@@ -33,7 +34,10 @@ DiMuonMassBiasMonitor::DiMuonMassBiasMonitor(const edm::ParameterSet& iConfig)
       MEFolderName_(iConfig.getParameter<std::string>("FolderName")),
       decayMotherName_(iConfig.getParameter<std::string>("decayMotherName")),
       distanceScaleFactor_(iConfig.getParameter<double>("distanceScaleFactor")),
-      DiMuMassConfiguration_(iConfig.getParameter<edm::ParameterSet>("DiMuMassConfig")) {}
+      DiMuMassConfiguration_(iConfig.getParameter<edm::ParameterSet>("DiMuMassConfig")) {
+  produces<DQMToken, edm::Transition::EndRun>("DQMGenerationDiMuonMassBiasMonitorRun");
+  produces<DQMToken, edm::Transition::EndLuminosityBlock>("DQMGenerationDiMuonMassBiasMonitorLumi");
```
Comment (makortel) [inline DQMOffline/Alignment/src/DiMuonMassBiasMonitor.cc:39]: @mmusich Did you have a specific reason to not use the existing `DQMToken` functionality in the DQM base classes?
https://github.com/cms-sw/cmssw/blob/f3a190193507fa09d12f82a397a350766f523629/DQMServices/Core/interface/DQMEDAnalyzer.h#L57-L58
https://github.com/cms-sw/cmssw/blob/f3a190193507fa09d12f82a397a350766f523629/DQMServices/Core/interface/DQMEDHarvester.h#L79-L95

Comment (mmusich) [reply DQMOffline/Alignment/src/DiMuonMassBiasMonitor.cc:39]: @makortel 

> Did you have a specific reason to not use the existing DQMToken functionality in the DQM base classes?

those were a relic from a previous stage of development in which this module was and `edm::EDAnalyzer` and not a `DQMEDHarvester`. That line of development was abandoned as the legacy modules are anyway deprecated.
I removed the useless statements in https://github.com/cms-sw/cmssw/pull/39217/commits/a5bab3b6c16061202c6fb8bbb6af0c0848d8a40a as part of  #39217

**Pull request merged**